src/term.rs

```diff
@@ -98,10 +111,47 @@ impl<VS: VersionSet> Term<VS> {
         }
     }
 
+    /// Check whether two terms are mutually exclusive.
+    ///
+    /// An optimization for the native implementation of checking whether the intersection of two sets is empty.
+    pub(crate) fn is_disjoint(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Positive(r1), Self::Positive(r2)) => r1.is_disjoint(r2),
+            (Self::Negative(r1), Self::Negative(r2)) => r1.union(r2) == VS::empty().complement(),
+            // If the positive term is a subset of the negative term, it lies fully in the region that the negative
+            // term excludes. Inversely, if there is a region outside the negative, they overlap in this region.
+            (Self::Positive(r1), Self::Negative(r2)) => r1.subset_of(r2),
+            (Self::Negative(r1), Self::Positive(r2)) => r2.subset_of(r1),
+        }
+    }
+
     /// Compute the union of two terms.
     /// If at least one term is negative, the union is also negative.
     pub(crate) fn union(&self, other: &Self) -> Self {
-        (self.negate().intersection(&other.negate())).negate()
```

Comment:
Inlining this code may be a micro optimization, saving a clone or two, but definitely seems worth doing.